share/getters/shrex.go

```diff
@@ -28,6 +28,23 @@ type ShrexGetter struct {
 	maxRequestDuration time.Duration
 }
 
+func NewShrexGetter(edsClient *shrexeds.Client, ndClient *shrexnd.Client, peerManager *peers.Manager) *ShrexGetter {
+	return &ShrexGetter{
+		edsClient:          edsClient,
+		ndClient:           ndClient,
+		peerManager:        peerManager,
+		maxRequestDuration: defaultMaxRequestDuration,
+	}
+}
+
+func (sg *ShrexGetter) Start(ctx context.Context) error {
+	return sg.peerManager.Start(ctx)
+}
+
+func (sg *ShrexGetter) Stop(ctx context.Context) error {
+	return sg.peerManager.Stop(ctx)
+}
```

Comment:
If ShrexGetter maintains the lifecycle of `peer.Manager`, it should also construct it, i.e. `peer.Manager` is a child of ShrexGetter but not a dependency. This also means that peer manager won't be part of FX dep graph, which is totally ok as the peer manager is a single purpose.

I don't wanna push this, and we can do it during later refactorings, so can make an issue

Reply:
Consider this is a feedback for https://github.com/celestiaorg/celestia-node/pull/1729